cptv-upload.py

```diff
@@ -0,0 +1,125 @@
+from deviceapi import DeviceAPI
+
+import argparse
+
+import os
+
+import json
+
+class CPTVUploader:
+    def __init__(self):
+        self.url = None        
+        self.source_dir = None
+        self.device_name = None
+        self.device_password = None
+
+    def process(self):
+        print('Uploading to  ', self.url)
+        api = DeviceAPI(self.url, devicename = self.device_name, password = self.device_password)
+
+        files = self._find_files_to_upload()
+        for file in files:
+            self._uploadfile(api, file)
+
+    def _uploadfile(self, api, filename):
+        props = self._readPropertiesFromFile(filename)
+
+        api.uploadrecording(filename, props)
+
+
+    def _uploader(self, queue, api):
+        while True:
+            
+            filename = queue.get()
+
+            if filename is None:
+                break
+
+            try:
+                _uploadfile(api, filename)
+
+            finally:
+                queue.task_done()
+
+    def _find_files_to_upload(self):
+        cptvfiles = list()
+
+        for root, dirs, files in os.walk(self.source_dir):
+            for file in files:
+                if file.lower().endswith(".cptv"):
+                    cptvfiles.append(os.path.join(root, file))
```

Comment:
There's nothing wrong with this but the `glob` package in the standard library will get you the same results in one line:
```python
cptvfiles = glob.glob(os.path.join(self.source_dir, '**', '*.cptv'), recursive=True)
```

`**` does recursive expansion